### PR TITLE
[WIP] Sorting projects based on "last_updated"

### PR DESCRIPTION
**Pull request by savvaspc**
I created a function in javascripts/projectsService.js that sorts the projects based on the "last_updated" field from Github API. For each project I check if it has a valid github repository, create a GET request in the form of "https://api.github.com/repos/{owner}/{repo}", and process it with AJAX. The timestamp is added as a new field in the project's object, and when all GET requests have been processed, they are sorted in descending order based on the timestamp. Projects that don't have a github repo are treated as if their timestamp is 0.

Of course this is not yet functional in the front end, but I believe it's a good first step. All I used is plain javascript and only a couple of functions from underscore.js.

Tries to implement #1142

### CODE REVIEW

File: javascripts/projectsService.js
Comment (ritwik12) [inline javascripts/projectsService.js]: @savvas94 There are multiple spaces before "_", Also your PR have lots of other errors which are shown in travis logs. please have a look on those.

**Review by ritwik12 — CHANGES_REQUESTED** [review]

**Comment by savvaspc** [comment]
@ritwik12 All the travis errors have been fixed. I'm currently working on the html button part, but I need guidance on how to present the final data on screen. I guess I could figure out how to do it myself, but I don't wanna break the already established way of doing it.

- [x] Write function that sends GET Requests, processes responds, and finally, sorts projects.
- [ ] Add html element (probably some "Sort by most recent" button) and pair it with a js event that runs the above code
- [ ] Render the sorted array into html

**Review by ritwik12 — APPROVED** [review]

**Review by shiftkey — CHANGES_REQUESTED** [review]
Holding this up for now because I feel this will only add to our pain in #261 and I'm not sure we can put together a client-side solution to that...

**Comment by shiftkey** [comment]
@savvas94 thanks for your interest in contributing to this, and apologies for the delay in getting over to review this.

Because of our already significant rate limiting issues, I don't believe we should support fetching the `last_updated` field from the GitHub API. Because of that, I focused on what we could do server-side. I didn't really feel like maintaining infrastructure to handle this, so it's basically sat unaddressed until recently, when GitHub Actions gave me a way to run infrastructure-esque tasks easily.

I've opened #1329 to track what we can when generating the static site, and have a demo involved in #1346 which makes `last-updated` available when we generate the site, which I think is the best way to tackle this. I'm going to update #1346 to include a simple "last updated" tooltip to show this works, but you're more than welcome to take another shot at #1142 once that's reviewed and merged.

**Pull request closed without merging**
